### PR TITLE
Get block by time stamp

### PR DESCRIPTION
**Pull request by enriavil1**

### CODE REVIEW

File: cmd/rpcdaemon/commands/erigon_block.go [hunk 1]
```diff
@@ -58,3 +58,72 @@ func (api *ErigonImpl) GetHeaderByHash(ctx context.Context, hash common.Hash) (*
...
+
+			return *block, nil
+		}
+
+		middleNumber = (highestNumber + lowestNumber) / 2
+		middleHeader = rawdb.ReadHeaderByNumber(tx, middleNumber)
+
+	}
+
+	return types.Block{}, nil
```
Comment (AlexeyAkhunov) [inline cmd/rpcdaemon/commands/erigon_block.go:127]: You should probably return the block with `highestNumber`. so in the case where precise timestamp is not found, you return the block just after that timestamp

**Comment by AlexeyAkhunov** [comment]
Can you also add a test (can you `eth_call_test.go` in the same package as a guide

**Comment by tjayrush** [comment]
From an accounting point of view you definitely want the block before the time stamp. If I ask for January 1, I want the block exactly then or before. You want transactions that were fully completed in the period.

**Comment by AskAlexSharov** [comment]
Do we wanna use blocksLRU for this blocks?

**Comment by AskAlexSharov** [comment]
@enriavil1 hi. Is it possible to use sort.Search here?

**Comment by enriavil1** [comment]
> @enriavil1 hi. Is it possible to use sort.Search here?

@AskAlexSharov  I believe sort.Search should work identically to the one I had written.

**Pull request merged**